src/example-types/facet.js

```diff
@@ -134,5 +138,76 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+
+    let missedValues = _.difference(
+      values,
+      _.map(
+        ({ name }) => (node.isMongoId ? _.toString(name) : name),
+        results.options
+      )
+    )
+
+    let getMissedValues = (node, missedValues) =>
+      node.isMongoId ? _.map(ObjectID, missedValues) : missedValues
+
+    let matchMissedValues = (node, missedValues) => ({
+      $match: { [node.field]: { $in: getMissedValues(node, missedValues) } },
+    })
+
+    if (!_.isEmpty(missedValues)) {
+      let searchMissedResult = await search(
+        _.compact([
+          matchMissedValues(node, missedValues),
+          { $group: { _id: `$${node.field}`, count: { $sum: 1 } } },
+          ...sortAndLimitIfNotSearching(node.optionsFilter, node.size),
+          ...lookupLabel(node),
+          _.get('label.fields', node) && projectStageFromLabelFields(node),
+          mapKeywordFilters(node),
+        ])
+      )
+      //stringify missedValues to avoid the bug when values are numeric values
+      let stillMissingValues = _.difference(
+        _.map(_.toString, missedValues),
+        _.map(x => _.toString(x[`${node.field}`]), searchMissedResult)
+      )
+      let stillMissingQueryFilter = {
+        [node.field]: { $in: getMissedValues(node, stillMissingValues) },
+      }
+      let stillMissingArgs = [
+        { $group: { _id: `$${node.field}` } },
+        ...lookupLabel(node),
+        _.get('label.fields', node) && projectStageFromLabelFields(node),
+      ]
+      let stillmissingResult = []
+      if (!_.isEmpty(stillMissingValues)) {
+        //use config to run runSearch(options, node, schema, filters, aggs)  function
+        stillmissingResult = await config
+          .getProvider(node)
+          .runSearch(
+            config.options,
+            node,
+            config.getSchema(node.schema),
+            stillMissingQueryFilter,
+            stillMissingArgs
+          )
+      }
+      let finalMissedReult = _.flow(
```

Comment:
`finalMissedReult` -> `finalMissedResult`

Reply:
a lot of these names are really similar - I think you're better off just inlining a bunch of these so it's easier to follow